.github/workflows/linux.yml

```diff
@@ -34,23 +40,54 @@ jobs:
         sudo apt-get update
         sudo apt-get install -y libopenblas-dev libatlas-base-dev liblapack-dev gfortran libgmp-dev libmpfr-dev libsuitesparse-dev ccache libmpc-dev
 
+    - name: Caching Python dependencies
+      uses: actions/cache@v2
+      id: cache
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip
+
     - name: Install Python packages
       run: |
         python -m pip install numpy setuptools wheel cython pytest pytest-xdist pybind11 pytest-xdist mpmath gmpy2 pythran ninja
         python -m pip install git+https://github.com/rgommers/meson.git@fix-cython-mixed-sources
 
+    - name:  Prepare compiler cache
+      id:    prep-ccache
+      shell: bash
+      run: |
+        mkdir -p "${CCACHE_DIR}"
+        echo "::set-output name=dir::$CCACHE_DIR"
+        echo "::set-output name=today::$(date -I)"
+
+    - name: Setup compiler cache
+      uses:  actions/cache@v2
+      id:    cache-ccache
+      with:
+        path: ${{ steps.prep-ccache.outputs.dir }}
+        key:  ccache-linux-${{ steps.prep-ccache.outputs.today }}
+        restore-keys: |
+          ccache-linux-
+
+    - name: Setup build
+      run: |
+        meson setup build --prefix=$PWD/${{ env.INSTALLDIR }}
+
     - name: Build SciPy
       run: |
-        meson setup build --prefix=$PWD/installdir
-        ninja -C build -j 2
+        ninja -C build
```

Comment:
Is there a reason you removed `-j 2`? I think we get 2 cores by default, so I'd expect it to be as fast as or faster than the Ninja default (which is `2*n + 2`).

Reply:
I did it on the recommendation of @leej3. Original comment here: https://github.com/HarshCasper/scipy-fork/commit/ae7cc2be6f4bdc28d23643d4e9194da051d2ee27#r56246908

Reply:
I was not aware of the Ninja default. Should I just revert it back?

Reply:
I have experimented with this, and it was optimal. The ninja default unfortunately isn't great. Doing it programmatically is less robust, because if you just ask for the number of cores in the most naive way you may get a large number (e.g., 64) and you need the number of cores that are available to the job, not the ones present on the machine.

I don't think this'll change any time soon, and it's documented that you get 2 cores. So yes, I'd suggest reverting this change.

Reply:
Ah, good to know.